pr-dhl-woocommerce/pr-dhl-woocommerce.php

```diff
@@ -296,23 +308,59 @@ public function get_pr_dhl_wc_product() {
 					} elseif ( $dhl_obj->is_dhl_deutsche_post() ) {
 						$this->shipping_dhl_product = new PR_DHL_WC_Product_Deutsche_Post();
 					}
-				} catch ( Exception $e ) {
+				}
+				catch ( Exception $e ) {
 					add_action( 'admin_notices', array( $this, 'environment_check' ) );
 				}
 			}
 
 			return $this->shipping_dhl_product;
 		}
 
+		/**
+		 * @return void
+		 */
+		public function get_product_editor() {
+			if ( ! is_null( $this->product_editor ) ) {
+				return;
+			}
+
+			try {
+				$dhl_obj = $this->get_dhl_factory();
+			}
+			catch ( Exception $e ) {
+				return;
+			}
+
+			if ( $dhl_obj->is_dhl_paket() ) {
+				$this->product_editor = new PR_DHL_WC_Product_Paket_Editor();
+			} elseif ( $dhl_obj->is_dhl_deutsche_post() ) {
+				$this->product_editor = new PR_DHL_WC_Product_Deutsche_Post_Editor();
+			}
+		}
```

Comment:
No need for duplicating this code, you can just add it to `get_pr_dhl_wc_product` function